time_execution/backends/threaded.py

```diff
@@ -24,6 +24,7 @@ def __init__(self, backend, backend_args=None, backend_kwargs=None,
             backend_args = tuple()
         if backend_kwargs is None:
             backend_kwargs = dict()
+        self.parent_thread = threading.current_thread()
```

Comment:
Are you assuming that the calling thread is always the main thread? This may break other use cases (e.g. instantiating the backend outside the main thread)...